### DISCUSSION TITLE
Dismissing Alerts

### DISCUSSION THREAD

**Discussion by rpatel8del** [pointfreeco/swift-composable-architecture#2036] [Q&A]
We have a timeout feature where we push the user out of the dashboard when we detect inactivity. 

If an alert is presented and we try to push the user out in all cases we successfully dismissed the alert but there's one case where we set the alert state to `nil` but the UI isn't updated. 

Can I get some pointers around how I could debug this?

**Comment by stephencelis** [comment]
@rpatel8del Do you have a repro you can share? It should be possible dismiss an alert by `nil`ing it out, but there may be something else going on here.

**Reply by rpatel8del** [reply]
I can't share my current repo, I'll try and simplify it in a different project and share that.

Also - I've found a way around it. I've had to isolate the `nil`ing of the state in one action and then push the user out (in a separate action) and this solves the issue. 
BUT  - does this mean I'm hitting some weird race condition when I try and update multiple properties in one action and only part of it is reflected in my UI?
